src/server.rs

```diff
@@ -83,12 +81,37 @@ impl Conmon for ConmonServerImpl {
     }
 }
 
-// Use the single threaded runtime to save rss memory
-#[tokio::main(flavor = "current_thread")]
-async fn main() -> Result<(), Error> {
-    let (shutdown_tx, shutdown_rx) = tokio::sync::oneshot::channel();
+fn main() {
+    // First, initialize the server so we have access to the config pre-fork.
+    let server = ConmonServerImpl::new().unwrap();
+
+    // We need to fork as early as possible, especially before setting up tokio.
+    // if we don't, the child will have a strange thread space and we're at risk of deadlocking.
+    // We also have to treat the parent as the child (as described in [1]) to ensure we don't
+    // interrupt the child's execution
+    // 1: https://docs.rs/nix/0.23.0/nix/unistd/fn.fork.html#safety
+    match unsafe{fork()} {
```

Comment:
This looks like it could use a `cargo fmt`

Reply:
It sure could :)